### PR TITLE
Add default constructors to MmrPeaks and PartialMmr

### PR DESCRIPTION
**Pull request by sergerad**
Add default constructors to MmrPeaks and PartialMmr and unit tests.

### CODE REVIEW

File: miden-crypto/src/merkle/mmr/partial.rs [hunk 1]
```diff
@@ -84,6 +84,16 @@ impl PartialMmr {
         Self { forest, peaks, nodes, track_latest }
     }
 
+    /// Returns a new [`PartialMmr`] instantiated with empty peaks and nodes and 0 leaves.
+    pub const fn empty() -> Self {
```
Comment (bobbinth) [inline miden-crypto/src/merkle/mmr/partial.rs:88]: Do we need this to be `const`? If not, then I'd probably prefer implementing `Default` trait from `PartialMmr`.

File: miden-crypto/src/merkle/mmr/peaks.rs [hunk 2]
```diff
@@ -55,6 +55,11 @@ impl MmrPeaks {
         Ok(Self { num_leaves, peaks })
     }
 
+    /// Returns new [`MmrPeaks`] instantiated from an empty vector of peaks and 0 leaves.
+    pub const fn empty() -> Self {
```
Comment (bobbinth) [inline miden-crypto/src/merkle/mmr/peaks.rs:59]: Same comment as above.

File: miden-crypto/src/merkle/mmr/tests.rs [hunk 3]
```diff
@@ -12,6 +12,23 @@ use crate::{
     merkle::{InOrderIndex, MerklePath, MerkleTree, MmrProof, NodeIndex, int_to_node},
 };
 
+#[test]
+fn tests_empty_mmr_peaks() {
+    let peaks = MmrPeaks::empty();
+    assert_eq!(peaks.num_peaks(), 0);
+    assert_eq!(peaks.num_leaves(), 0);
+}
```
Comment (bobbinth) [inline miden-crypto/src/merkle/mmr/tests.rs:20]: Another test could be to construct an empty `Mmr` and then get peaks from it and make sure these are equal to empty `MmrPeaks`.

**Review by bobbinth — COMMENTED** [review]
Looks good! Thank you! I left a couple of comments inline - the main one is about potentially using `Default` impl instead of explicit `empty()` methods.

File: miden-crypto/src/merkle/mmr/partial.rs (hunk 1 above)
Comment (sergerad) [reply miden-crypto/src/merkle/mmr/partial.rs:88]: The const is not necessary. Wasn't sure if `Default` impl was appropriate here - will do now

File: miden-crypto/src/merkle/mmr/tests.rs (hunk 3 above)
Comment (sergerad) [reply miden-crypto/src/merkle/mmr/tests.rs:20]: The test below does that - unless I'm misunderstanding your comment:
```rust
fn test_empty_partial_mmr() {
    let mmr = PartialMmr::empty();
    assert_eq!(mmr.num_leaves(), 0);
    assert_eq!(mmr.forest(), 0);
    assert_eq!(mmr.peaks(), MmrPeaks::empty());
```

Comment (bobbinth) [reply miden-crypto/src/merkle/mmr/tests.rs:20]: Ah - sorry - missed it!

**Comment by bobbinth** [comment]
btw, if this is something we need right now, we can make this PR against `main` and I can release a patch version (since this is a non-breaking change).

**Comment by sergerad** [comment]
> btw, if this is something we need right now, we can make this PR against `main` and I can release a patch version (since this is a non-breaking change).

Good to know. But its not critical - just helps tidy the faucet a little bit. Also need to update the `ChainMmr` type to use these so will make a PR to base (assuming its there) after this one.

On the note of non-breaking change - I shouldn't update the changelog like I have.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

> Good to know. But its not critical - just helps tidy the faucet a little bit. Also need to update the ChainMmr type to use these so will make a PR to base (assuming its there) after this one.

If we merge into `next` it may take some time to propagate it to `miden-vm` -> `miden-base` -> `miden-node`. But If we make a patch release, it'll be immediately available there.

**Comment by sergerad** [comment]
> Looks good! Thank you!
> 
> > Good to know. But its not critical - just helps tidy the faucet a little bit. Also need to update the ChainMmr type to use these so will make a PR to base (assuming its there) after this one.
> 
> If we merge into `next` it may take some time to propagate it to `miden-vm` -> `miden-base` -> `miden-node`. But If we make a patch release, it'll be immediately available there.

This is the case because some `next` branches point to a tag/release rather than `next` right?

Base points at 0.8: https://github.com/0xMiden/miden-base/blob/main/Cargo.toml#L46
Client points at next: https://github.com/0xMiden/miden-client/blob/next/Cargo.toml#L24

Ultimately would like to push to one repo's `next` branch and then use that code in another repo's `next` branch without any releases. Is that possible today?

**Comment by bobbinth** [comment]
In `base` -> `node` -> `client` this is how it works. But in `miden-crypto` -> `miden-vm` the dependencies are based on released versions. Notably, current `next` in `miden-vm` has some breaking changes that which are not yet propagated to `miden-base`.

**Comment by sergerad** [comment]
Ah ok thats interesting. That seems like another example of how the `next` branch workflow tends to "invert" the relationship between trunk and release. Whereas if `main` was trunk, then `miden-vm` would have had to have made a feature or release branch to deviate. That way the trunks could continue to be developed with latest changes reaching each repo without the need of a release.
